backend/controller/controller.go

```diff
@@ -1566,6 +1570,36 @@ func (s *Service) catchAsyncCall(ctx context.Context, logger *log.Logger, call *
 	return nil
 }
 
+// fails async calls that have had their leases reaped
+func (s *Service) reapAsyncCalls(ctx context.Context) (nextInterval time.Duration, err error) {
+	tx, err := s.dal.Begin(ctx)
+	if err != nil {
+		return 0, connect.NewError(connect.CodeInternal, fmt.Errorf("could not start transaction: %w", err))
+	}
+	defer tx.CommitOrRollback(ctx, &err)
+
+	limit := 20
+	calls, err := tx.GetZombieAsyncCalls(ctx, 20)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get zombie async calls: %w", err)
+	}
+	for _, call := range calls {
+		callResult := either.RightOf[[]byte]("async call lease expired")
+		_, err := s.dal.CompleteAsyncCall(ctx, call, callResult, func(tx *dal.Tx, isFinalResult bool) error {
+			return s.finaliseAsyncCall(ctx, tx, call, callResult, isFinalResult)
+		})
+		if err != nil {
+			return 0, fmt.Errorf("failed to complete zombie async call: %w", err)
+		}
+		// TODO: telemetery
```

Comment:
Let's do the telemetry in this PR, or file a ticket.

Reply:
Added 👍